cmd/server/api/handler.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"net/http"
+	"time"
+)
+
+// Define an interface for HTTP client operations
+type HttpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Create a wrapper for the net/http client
+type HttpClientWrapper struct {
+	Client *http.Client
+}
+
+func (h *HttpClientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return h.Client.Do(req)
+}
+
+type ApiHandlerInterface interface {
+	PostRpcHandler(w http.ResponseWriter, r *http.Request)
+}
+
+type ApiHandler struct {
+	RpcUrl     string
+	HttpClient HttpClientInterface // Inject the HTTP client interface here
+}
+
+type ApiHandlerOption func(*ApiHandler)
+
+func WithHttpClient(client HttpClientInterface) ApiHandlerOption {
```

Comment:
I don't really understand the need for this. Do we have an  idea for ApiHandler to work without httpClient?Where? If not why do we need explicit functions for this?